### PR TITLE
TTV Wizard: Import data step

### PR DESCRIPTION
**Pull request by Jonathan-Rosenberg**
## What's in this PR?
1.  Breaking up and extracting the import component into multiple components.
2. Changing the import button to use the extracted components.
3. Creating an "import data" step using the extracted components.
4. Using the "import data" step in the wizard. 
5. Formatting the files affected by the changes.

### Wizard notes
The files that relate to the changes in the wizard are:
* [import_data_wizard_step.jsx](https://github.com/treeverse/lakeFS/compare/feature/wizard-step-wrapper...feature/wizard-import-data-step?expand=1#diff-9e30b473dc585453dcebeb3ef02a981ace394e8783d76327ec33c224cd9649cb)
* [spark_quickstart_wizard.jsx](https://github.com/treeverse/lakeFS/compare/feature/wizard-step-wrapper...feature/wizard-import-data-step?expand=1#diff-94ec978f51e03f0465892e952c163518b1b4c5f995162211445dd82b8253bc93)

## How was this tested
1. Made sure that the normal functionality of the import button works as expected- imported files, added commit message, added path...
2. Imported files through the Wizard to the newly initialized repository's main branch.

## Visuals
Screen Shot 2022-07-12 at 13 43 37
Screen Shot 2022-07-12 at 13 45 28
Screen Shot 2022-07-12 at 13 45 36
Screen Shot 2022-07-12 at 13 45 44
Screen Shot 2022-07-12 at 13 48 50

Closes [#3553](https://github.com/treeverse/lakeFS/issues/3553)

### CODE REVIEW

**Comment by arielshaqed** [comment]
`ETOOMANYREVIEWERS`, removing myself as I doubt I have much to add :-) (other than noisy disagreement with all other reviewers of course).  Please re-add me for any _specific_ aspects (preferably not entirely frontend...).

**Review by johnnyaug — APPROVED** [review]
Great refactor!
Minor comments

File: webui/src/pages/repositories/wizard/import_data_wizard_step.jsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+    return (
+        <>
+            <h2 className={"wizard-step-header"}>Import Data</h2>
+            {
+                importPhase === ImportPhase.NotStarted &&
+                <ImportForm
+                    pathStyle={{'minWidth': '25%'}}
+                    sourceRef={sourceRef}
+                    updateSrcValidity={(isValid) => setIsSourceValid(isValid)}
```
[the remark this answers is not in the document]
Comment (Jonathan-Rosenberg) [reply webui/src/pages/repositories/wizard/import_data_wizard_step.jsx:50]: I named it `updateSrcValidity` so it won't be confused with a state (that doesn't exist) within the component (`ImportForm`) itself.
You think that it's better to change it?

File: webui/src/pages/repositories/repository/objects.jsx [hunk 2]
```diff
@@ -17,187 +17,128 @@ import {useAPI, useAPIWithPagination} from "../../../lib/hooks/api";
...
-                repo.id,
-                reference.id
+                repoId,
+                referenceId
             );
             onDone();
         } catch (error) {
-            setImportState({...initialState, error})
+            setImportError(error);
             throw error
```
[the remark this answers is not in the document]
Comment (Jonathan-Rosenberg) [reply webui/src/pages/repositories/repository/objects.jsx:89]: true

File: webui/src/pages/repositories/services/import_data.jsx [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+    updateImportState(importStatusUpdate);
+    do {
+        const response = await ranges.createRange(repoId, sourceRef, after, prependPath, paginationResp.continuation_token);
+        rangeArr.push(response.range);
+        paginationResp = response.pagination;
+        after = paginationResp.last_key;
+        sum += response.range.count;
+        importStatusUpdate.numObj = sum
+        updateImportState(importStatusUpdate);
+    } while (paginationResp.has_more);
```
Comment (eden-ohana) [inline webui/src/pages/repositories/services/import_data.jsx:36]: Did you test the import with pagination?

**Review by eden-ohana — CHANGES_REQUESTED** [review]

File: webui/src/pages/repositories/repository/objects.jsx [hunk 4]
```diff
@@ -17,187 +17,129 @@ import {useAPI, useAPIWithPagination} from "../../../lib/hooks/api";
...
-                repo.id,
-                reference.id
+                repoId,
+                referenceId
             );
             onDone();
         } catch (error) {
-            setImportState({...initialState, error})
-            throw error
+            setImportError(error);
```
[the remark this answers is not in the document]
Comment (Jonathan-Rosenberg) [reply webui/src/pages/repositories/repository/objects.jsx:88]: You're right. I actually didn't manage to get to it, but there's a missing "phase" which is `ImportPhase.Failed`.
I'll add it.
Thanks!

File: webui/src/pages/repositories/services/import_data.jsx [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+import Row from "react-bootstrap/Row";
+import Col from "react-bootstrap/Col";
+import {LinearProgress} from "@mui/material";
+import React from "react";
+import {Link} from "react-router-dom";
+import Button from "react-bootstrap/Button";
+import Alert from "react-bootstrap/Alert";
+import Form from "react-bootstrap/Form";
+
+const ImportPhase = {
```
[the remark this answers is not in the document]
Comment (Jonathan-Rosenberg) [reply webui/src/pages/repositories/services/import_data.jsx:11]: 😺

File: webui/src/pages/repositories/repository/objects.jsx [hunk 6]
```diff
@@ -17,187 +17,129 @@ import {useAPI, useAPIWithPagination} from "../../../lib/hooks/api";
...
-        error: null,
-        done: false,
-        numObj: 0,
-        isSourceValid: null
-    }
-    const [importState, setImportState] = useState(initialState)
+const ImportModal = ({config, repoId, referenceId, referenceType, path = '', onDone, onHide, show = false}) => {
+    const [importPhase, setImportPhase] = useState(ImportPhase.NotStarted);
+    const [numberOfImportedObjects, setNumberOfImportedObjects] = useState(0);
+    const [isSourceValid, setIsSourceValid] = useState(false);
```
[the remark this answers is not in the document]
Comment (Jonathan-Rosenberg) [reply webui/src/pages/repositories/repository/objects.jsx:40]: It's inside the [`ImportFormat` component](https://github.com/treeverse/lakeFS/blob/3229ba3c345d783212d8a45ab3b8b70879334038/webui/src/pages/repositories/services/import_data.jsx#L151).
It actually belongs to the `ExecuteImportButton` component. I'll refactor.
Thanks again.

File: webui/src/pages/repositories/services/import_data.jsx (hunk 3 above)
Comment (Jonathan-Rosenberg) [reply webui/src/pages/repositories/services/import_data.jsx:36]: yes

File: webui/src/pages/repositories/services/import_data.jsx [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+                        sourceRef,
+                        destRef,
+                        repoId,
+                        importBranch,
+                        path,
+                        commitMsgRef,
+                        updateSrcValidity,
+                        shouldAddPath = false
+                    }) => {
+    const [isSourceValid, setIsSourceValid] = useState(false);
```
Comment (eden-ohana) [inline webui/src/pages/repositories/services/import_data.jsx:138]:
```suggestion
    const [isSourceValid, setIsSourceValid] = useState(null);
```

File: webui/src/pages/repositories/services/import_data.jsx [hunk 8]
```diff
@@ -0,0 +1,194 @@
...
+    const storageNamespaceValidityRegexStr = config.blockstore_namespace_ValidityRegex;
+    const storageNamespaceValidityRegex = RegExp(storageNamespaceValidityRegexStr);
+    const checkSourceURLValidity = () => {
+        setIsSourceValid(storageNamespaceValidityRegex.test(sourceRef.current.value));
```
Comment (eden-ohana) [inline webui/src/pages/repositories/services/import_data.jsx:142]: change the state to enable to ExecuteImportButton

**Review by eden-ohana — CHANGES_REQUESTED** [review]
Thanks! some comments
* The import button stays disabled even when you enter a valid source URI
* The message invalid source URI shows before you type anything
* I think the error message should appear on the import form page, for consistency with the other forms error - repository, branch. and if a user enters an incorrect source URI he needs the exit the import page and gets back to it again - instead of just filing the form again.

File: webui/src/pages/repositories/services/import_data.jsx [hunk 9]
```diff
@@ -0,0 +1,198 @@
...
+                    </p>
+                </div>
+            }
+        </Col>
+    </Row>);
+}
+const ExecuteImportButton = ({isEnabled, importPhase, importFunc}) => {
+    return (
+        <Button variant="success"
+                disabled={!isEnabled}
```
Comment (eden-ohana) [inline webui/src/pages/repositories/services/import_data.jsx:114]: You need to update the state when the import is in progress and completed to disable the import button. Currently, you can press it during and at the of the import and get an error.

**Review by eden-ohana — APPROVED** [review]
Thanks!

File: webui/src/pages/repositories/services/import_data.jsx (hunk 9 above)
Comment (Jonathan-Rosenberg) [reply webui/src/pages/repositories/services/import_data.jsx]: Thanks!!!

**Pull request merged**